generic/generic_haiku_app_secondary_arch-1.2.3.recipe

```diff
@@ -0,0 +1,87 @@
+SUMMARY="Short description of ProjectX"
+DESCRIPTION="Long ProjectX description.
+Spanning several paragraphs, maybe
+ - containing
+ - bullet points etc.
+You can also have really long lines that do not fit in 100 characters and use a backslash to \
+wrap the line."
+HOMEPAGE="https://homepage/of/projectx.org"
+COPYRIGHT="2014 Developer name
+	2016 Another developer's name"
+LICENSE="MIT
+	GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/HaikuArchives/ProjectX/archive/projectx-$portVersion.tar.gz"
+CHECKSUM_SHA256="000000000000000000000000000000000"
+SOURCE_DIR="$portVersionedName"
+PATCHES="projectx-$portVersion.patchset"
+ADDITIONAL_FILES="projectx.rdef"
+
+ARCHITECTURES="?all !x86_gcc2"
+SECONDARY_ARCHITECTURES="?x86"
+
+GLOBAL_WRITABLE_FILES="
+	settings/projectx.rc keep-old
+	"
+
+PROVIDES="
+	projectx$secondaryArchSuffix = $portVersion
+	app:projectx = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+defineDebugInfoPackage projectx$secondaryArchSuffix \
+	$appsDir/ProjectX
+
+BUILD()
+{
+	make $jobArgs OBJ_DIR=objects
+}
+
+INSTALL()
+{
+	install -d $appsDir
+	install -t $appsDir objects/ProjectX
+
+	# Create the extra attributes
+	local APP_SIGNATURE="application/x-vnd.ProjectX"
+	local MAJOR="`echo "$portVersion" | cut -d. -f1`"
+	local MIDDLE="`echo "$portVersion" | cut -d. -f2`"
+	local MINOR="`echo "$portVersion" | cut -d. -f3`"
+	local MINOR="0"
+	local LONG_INFO="$SUMMARY"
+
+	# Add them to the rdef file
+	sed \
+		-e "s|@APP_SIGNATURE@|$APP_SIGNATURE|" \
+		-e "s|@MAJOR@|$MAJOR|" \
+		-e "s|@MIDDLE@|$MIDDLE|" \
+		-e "s|@MINOR@|$MINOR|" \
+		-e "s|@LONG_INFO@|$LONG_INFO|" \
+		$portDir/additional-files/projectx.rdef.in > projectx.rdef
+
+	# Add the resources to the application
+	addResourcesToBinaries projectx.rdef $appsDir/ProjectX
+
+	# Set the acceptable mimetypes
+	mimeset -f $appsDir/ProjectX
+
+	# Create a Deskbar icon
+	addAppDeskbarSymlink $appsDir/ProjectX
+}
+
+TEST()
+{
+	make check
```

Comment:
I don't think that makefile-engine has a `check` target, or does it? If not, I would just remove the TEST function from this.

Reply:
Done, thanks.